crates/wasmtime/src/func.rs

```diff
@@ -1479,19 +1475,27 @@ impl Caller<'_> {
                 return None;
             }
             let instance = InstanceHandle::from_vmctx(self.caller_vmctx);
-            let export = match instance.lookup(name) {
-                Some(Export::Memory(m)) => m,
-                _ => return None,
-            };
-            // Our `Weak` pointer is used only to break a cycle where `Store`
-            // stores instance handles which have this weak pointer as their
-            // custom host data. This function should only be invoke-able while
-            // the `Store` is active, so this upgrade should always succeed.
-            debug_assert!(self.store.upgrade().is_some());
-            let handle =
-                Store::from_inner(self.store.upgrade()?).existing_instance_handle(instance);
-            let mem = Memory::from_wasmtime_memory(export, handle);
-            Some(Extern::Memory(mem))
+            match instance.lookup(name) {
+                Some(Export::Memory(m)) => {
+                    // Our `Weak` pointer is used only to break a cycle where `Store`
+                    // stores instance handles which have this weak pointer as their
+                    // custom host data. This function should only be invoke-able while
+                    // the `Store` is active, so this upgrade should always succeed.
+                    debug_assert!(self.store.upgrade().is_some());
+                    let handle =
+                        Store::from_inner(self.store.upgrade()?).existing_instance_handle(instance);
+                    let mem = Memory::from_wasmtime_memory(m, handle);
+                    Some(Extern::Memory(mem))
+                }
+                Some(Export::Function(f)) => {
+                    debug_assert!(self.store.upgrade().is_some());
+                    let handle =
+                        Store::from_inner(self.store.upgrade()?).existing_instance_handle(instance);
```

Comment:
Could this be refactored a bit to avoid duplicating the `let handle = ...`?

Something like:

```rust
let export = instance.lookup(name)?:
let handle = ...;
Some(match export {
    Export::Memory(m) => Export::Memory(Memory::from_wasmtime_memory(m, handle)),
    Export::Function(f) => /* ... */,
    _ => return None,
})